File: Arrays & HashMap/Two Sum/Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = defaultdict(int)

        for idx, num in enumerate(nums):
            diff = target - num

            if diff in map:
                return [idx, map[diff]]
            
            map[num] = idx
        
        return []

'''
Time Complexity: O(n)
Space Complexity: O(n)    
'''

        

File: Arrays & HashMap/Two Sum/Two Sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()

        nums.forEachIndexed { idx, num -> 
            val diff = target - num
            if (map.containsKey(diff)) {
                return intArrayOf(idx, map[diff]!!)
            }
            map[num] = idx
        }
        return intArrayOf(-1, -1)
    }
}

//Time Complexity: O(n)
//Space Complexity: O(n)